extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -78,6 +81,20 @@ export class AnalysesResultsManager {
     return [...this.analysesResults];
   }
 
+  /**
+   * Prepares a directory for storing analysis results for a single query run.
+   * This directory initially contains only a timestamp file, which will be
+   * used by the query history manager to determine when the directory
+   * should be deleted.
+   *
+   * @param queryName The name of the query that was run.
+   */
+  public async prepareDownloadDirectory(queryName: string): Promise<void> {
+    // Prepare the storage directory.
+    const artifactStorageDir = path.join(this.storagePath, `${queryName}-${nanoid()}`);
```

Comment:
Not sure this is a useful comment but just something to think about.. We seem to be going down an implementation path where various parts of the code control how query history data is stored. For example, in this case, the `AnalysesResultsManager` decides to follow the <queryName>-<id> pattern. But it could decide to follow a different pattern and (I imagine) that would make other bits of the code unhappy. 

I've not thought about query history nearly as much as you have so I could be wrong, but it feels like the responsibility and knowledge around query history should be in a single place.

Perhaps what we're thinking here is overall data of a query, not "historical" data only. Maybe instead of a `storagePath` we could to be giving these objects a class/function that will tell them where to store their data.